### DISCUSSION TITLE
WpfDrawingSettings  PixelWidth PixelHeight Have no effect

### DISCUSSION THREAD

**Discussion by dongzhaosheng73** [ElinamLLC/SharpVectors#258] [Q&A]
```cs
var setting = new WpfDrawingSettings() { IncludeRuntime = true, TextAsGeometry = true };
setting.PixelWidth = 100;
setting.PixelHeight = 100;
using (FileSvgReader converter = new FileSvgReader(setting))
{
	var draw = new DrawingImage(converter.Read(net.Stream));
	draw.Freeze();
	return draw;
}            
```
The DrawingImage generated is not width = 100 height = 100. The DrawingImage is still the original size of the vector image. Is there any problem with my setting?

**Comment by paulushub** [comment]
Currently, we do not transform the `DrawingGroup`, so this feature only works when creating static images.
You can try rescaling the output of the `converter.Read(net.Stream)`.

**Comment by dongzhaosheng73** [comment]
> Currently, we do not transform the `DrawingGroup`, so this feature only works when creating static images.
> You can try rescaling the output of the `converter.Read(net.Stream)`.

I want to make sure if changing the size of DrawingImage can reduce the memory usage?
I tried to change the DrawingGroup directly by modifying the Transform, and the size of the DrawingImage did change, but it didn't seem to decay at all when it was filled into the control. Memory is not reduced.

```cs
 var setting = new WpfDrawingSettings() { IncludeRuntime = true, TextAsGeometry = true,EnsureViewboxSize = false};
          
using (FileSvgReader converter = new FileSvgReader(setting))
{
	var drawGroup = converter.Read(net.Stream);
	drawGroup.Transform = new ScaleTransform(0.1, 0.1);
	var draw = new DrawingImage(drawGroup);
	draw.Freeze();
	return draw;
}
```

**Comment by paulushub** [comment]
> I want to make sure if changing the size of DrawingImage can reduce the memory usage?

I am not sure it will. `DrawingImage` is basically still a vector drawing implementation of the `ImageSource`. It is not the actual rendered image, just the source of the image and applying a transformation only add a projection definition.
